src/cmdstan/command.hpp

```diff
@@ -125,32 +126,6 @@ int command(int argc, const char *argv[]) {
   if (parser.help_printed())
     return return_codes::OK;
 
-  int num_threads = get_arg_val<int_argument>(parser, "num_threads");
-  // Need to make sure these two ways to set thread # match.
-  int env_threads = stan::math::internal::get_num_threads();
-  if (env_threads != num_threads) {
-    if (env_threads != 1) {
-      std::stringstream thread_msg;
-      thread_msg << "STAN_NUM_THREADS= " << env_threads
-                 << " but argument num_threads= " << num_threads
-                 << ". Please either only set one or make sure they are equal.";
-      throw std::invalid_argument(thread_msg.str());
-    }
-  }
-  stan::math::init_threadpool_tbb(num_threads);
-
-  unsigned int num_chains = 1;
-  auto user_method = parser.arg("method");
-  if (user_method->arg("sample")) {
-    num_chains
-        = get_arg_val<int_argument>(parser, "method", "sample", "num_chains");
-    if (num_chains > 1)
-      validate_multi_chain_config(parser.arg("method"));
-  }
```

Comment:
It appears that the `validate_multi_chain_config` function is no longer called anywhere. Either its functionality was replaced (in which case we can delete it from the helpers file) and I can't find it, or we lost this check somewhere and need to put it back

Reply:
good catch.  we should have this - moved too much stuff around.  added back.

Reply:
I noticed another function in `helper` called `allow_multichain` which seems to do mostly the same thing (+ some pathfinder logic), but doesn't look exactly the same

Reply:
that's used by another helper function - `get_num_chains`.  added back `validate_multi_chain_config` call, also using `get_num_chains` since with pathfinder,  `num_paths` also determines `num_chains`.